internal/pkg/service/stream/storage/level/local/encoding/pipeline.go

```diff
@@ -207,13 +209,8 @@ func (w *pipeline) WriteRecord(timestamp time.Time, values []any) error {
 		return errors.New(`writer is closed`)
 	}
 
-	// Check values count
-	if len(values) != len(w.slice.Columns) {
-		return errors.Errorf(`expected %d columns in the row, given %d`, len(w.slice.Columns), len(values))
-	}
```

Comment:
good question, I forgot to explain it.

It can't happen anymore, it can't be simulated, because it's now solved within one unit, so the number of columns is always the same.